### ISSUE TITLE
With Mode-icons, the clock is not aligned anymore

### ISSUE THREAD

**Issue by ReneFroger** [Malabarba/smart-mode-line#176]
My smart mode line is configured as followed

    (use-package smart-mode-line
        :ensure t 
        :config
        (setq display-time-default-load-average nil)

        (setq-default mode-line-buffer-identification
                        (let ((orig (car mode-line-buffer-identification)))
                        `(:eval (cons (concat ,orig " " (abbreviate-file-name default-directory))
                                        (cdr mode-line-buffer-identification)))))
        ;; -----------------------------------------------------------------------------
        ;; Display time in mode line 
        ;; -----------------------------------------------------------------------------
        (defun enable-time ()
            (interactive)
            (setq display-time-24hr-format t)
            (setq display-time-day-and-date t)
            (display-time)
            ;; Align the time to right
            (setq global-mode-string (remove 'display-time-string global-mode-string))
            (setq mode-line-end-spaces
                (list (propertize " " 'display '(space :align-to (- right 17)))
                        'display-time-string))
            )   
        (setq display-time-default-load-average nil) 
        (add-hook 'after-init-hook 'display-time) 
        (setq display-time-24hr-format t)
        (setq display-time-day-and-date t)
        (setq sml/no-confirm-load-theme t)
        (setq sml/name-width 60)
        (sml/setup)
    )

Yesterday I discovered (https://github.com/ryuslash/mode-icons)[Mode icons]. But I also noticed it moves the time display to left, instead right in the corner.

So I thought this was caused when Mode Icons shortened the text of the major mode to text, then the time display got moved because the text is also shortened.

Instead asking of a fix, I thought this was perhaps initially caused by a bad design. Is there something like a Emacs fixed value which is persistent, that causes the time display to always on the right (like in Windows, I render my Emacs as full screen), instead moving all the way on the mode line?

**Comment by Malabarba** [comment]
>  Is there something like a Emacs fixed value which is persistent, that causes the time display to always on the right (like in Windows, I render my Emacs as full screen), instead moving all the way on the mode line?

No, not that I am aware. I looked this up way back when I implemented the fixed-width feature. But if you know of anything useful, I'm all years. :+1:
